### PR TITLE
SONiC yang models for AAA and TACACS features

### PR DESCRIPTION
**Pull request by ArthiSivanantham**
Signed-off-by: Arthi Sivanantham <arthi_sivanantham@dell.com>

#### Why I did it
SONiC YANG model support for AAA and TACACS features.

#### How I did it
Defined various AAA and TACACS YANG containers and lists based on config-DB schema.

#### How to verify it
Successful build of the following packages:
make target/python-wheels/sonic_yang_models-1.0-py3-none-any.whl
make target/python-wheels/sonic_yang_mgmt-1.0-py3-none-any.whl

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012

#### Description for the changelog
SONiC Yang model support for AAA and TACACS

### CODE REVIEW

File: src/sonic-yang-models/tests/yang_model_tests/tests/aaa.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+    "AAA_TEST": {
+        "desc": "Configure an authentication type in AAA table."
+    },
+    "AAA_TEST_WRONG_TYPE": {
+        "desc": "Configure a wrong type in AAA table.",
+	"eStrKey": "InvalidValue"
```
Comment (jleveque) [inline src/sonic-yang-models/tests/yang_model_tests/tests/aaa.json:7]: Seems to be alignment issues in some of these files due to a mixture of tabs and spaces. Please replace tabs with spaces everywhere.

**Review by jleveque — CHANGES_REQUESTED** [review]

File: src/sonic-yang-models/tests/yang_model_tests/tests/aaa.json (hunk 1 above)
Comment (ArthiSivanantham) [reply src/sonic-yang-models/tests/yang_model_tests/tests/aaa.json]: Addressed

**Comment by anshuv-mfst** [comment]
@ArthiSivanantham -  please comment out in input vlan to pass build checks.

**Comment by anshuv-mfst** [comment]
@renukamanavalan - can you please review and approve

File: src/sonic-yang-models/yang-models/sonic-system-aaa.yang [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+                leaf login {
+                    type string;
+                    description "AAA authentication methods - local/tacacs+";
+                    default "local";
+                }
+
+                leaf failthrough {
+                    type boolean;
+                    description "When set to true, authentication is attempted on next configured server/local in the list upon failure.";
```
Comment (renukamanavalan) [inline src/sonic-yang-models/yang-models/sonic-system-aaa.yang:30]: default?

File: src/sonic-yang-models/yang-models/sonic-system-aaa.yang [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                }
+                
+                leaf debug {
+                    type boolean;
+                    description "Enable/disable AAA debugging";
```
Comment (renukamanavalan) [inline src/sonic-yang-models/yang-models/sonic-system-aaa.yang:35]: default?

File: src/sonic-yang-models/yang-models/sonic-system-tacacs.yang [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+//    }
+
+    revision 2021-04-15 {
+        description "Initial revision.";
+    }
+
+    typedef auth_type_enumeration {
+        type enumeration {
+            enum pap;
+            enum chap;
```
Comment (renukamanavalan) [inline src/sonic-yang-models/yang-models/sonic-system-tacacs.yang:43]: Does SONiC support chap & mschap?

Comment (venkatmahalingam) [reply src/sonic-yang-models/yang-models/sonic-system-tacacs.yang:43]: As per the below HLD, chap & mschap are supported.
https://github.com/Azure/SONiC/blob/master/doc/aaa/TACACS%2B%20Authentication.md

File: src/sonic-yang-models/yang-models/sonic-system-aaa.yang (hunk 2 above)
Comment (ArthiSivanantham) [reply src/sonic-yang-models/yang-models/sonic-system-aaa.yang:30]: Added default false

File: src/sonic-yang-models/yang-models/sonic-system-aaa.yang (hunk 3 above)
Comment (ArthiSivanantham) [reply src/sonic-yang-models/yang-models/sonic-system-aaa.yang:36]: Added default false

File: src/sonic-yang-models/yang-models/sonic-system-tacacs.yang [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+                            path "/lag:sonic-portchannel/lag:PORTCHANNEL/lag:PORTCHANNEL_LIST/lag:name";
+                        }
+                        type leafref {
+                            path "/vlan:sonic-vlan/vlan:VLAN/vlan:VLAN_LIST/vlan:name";
+                        }
+                        type leafref {
+                            path "/loopback:sonic-loopback-interface/loopback:LOOPBACK_INTERFACE/loopback:LOOPBACK_INTERFACE_LIST/loopback:name";
+                        }
+                        //type leafref {
+                        // path "/mgmt-port:sonic-mgmt-port/mgmt-port:MGMT_PORT/mgmt-port:MGMT_PORT_LIST/mgmt-port:name";
```
Comment (venkatmahalingam) [inline src/sonic-yang-models/yang-models/sonic-system-tacacs.yang]: Please uncomment these lines as MGMT YANG PR is already merged.

**Review by venkatmahalingam — APPROVED** [review]

**Comment by ArthiSivanantham** [comment]
@ jleveque - All comments are addressed. Can you please re-review  and approve ? Thanks.

**Review by jleveque — APPROVED** [review]
Looks good from my perspective. @renukamanavalan to review, as well.

**Review by renukamanavalan — APPROVED** [review]

**Pull request merged**